### DISCUSSION TITLE
Getting "Laravel\Nova\Fields\BelongsTo::buildMorphableQuery does not exist" on a Polymorphic resource

### DISCUSSION THREAD

**Discussion by lukejwatts** [laravel/nova-issues#6474] [Q&A]
- Laravel Version: 9.52.16
- Nova Version: Version: 4.22.2 (Silver Surfer)
- PHP Version:  8.2.17
- Database Driver & Version: MySQL - Ver 8.0.36 for Win64 on x86_64 (MySQL Community Server - GPL)
- Operating System and Version: Windows 10
- Browser type and version: MS Edge - Version 126.0.2592.102 (Official build) (64-bit)
- Reproduction Repository: Todo

### Description:

I have a polymorhic table which has a belongs to relationship on it also. When I attempt to select the resource (project_phaseable_type) from the dropdown I get the following error:

![image](https://github.com/user-attachments/assets/32c1cb97-d11a-4609-bcc6-8dbcb678f3fd)
 
I'm assuming it's because of the BelongsTo field in my fields array? Or I've set something up wrong (Please see the code snippets below for migrations, model and resource definitions)

Am I using the wrong morphTo/Many etc? The models relationships all work as expected for my use case which is many-to-many from the example, but it's actually a one-to-one relationship in practice.

## Polymorphic Migration Schema

`create_project_phases_table` migration:

```php
Schema::create('project_phases', function (Blueprint $table) {
    $table->id();
    $table->unsignedBigInteger('phase_id');
    $table->morphs('project_phaseable');
    $table->timestamps();

    // Ensures we can only have one phase per "project_phaseable_type"
    $table->unique(['phase_id', 'project_phaseable_type', 'project_phaseable_id'], 'project_phaseable_polymorphic_unique');

    $table->foreign('phase_id')->references('id')->on('phases');
});
```
### ProjectPhase model

```php
class ProjectPhase extends Model
{
    use HasFactory;

    /**
     * Get the parent project_phaseable model (family, estate, property).
     */
    public function projectPhaseable()
    {
        return $this->morphTo();
    }

    public function phase(): BelongsTo
    {
        return $this->belongsTo(Phase::class, 'phase_id', 'id');
    }
}
```

### ProjectPhase resource (fields)

```php
public function fields(NovaRequest $request)
    {
        return [
            ID::make()->sortable(),
            BelongsTo::make('Phase', 'projectPhaseable', Phase::class),
            MorphTo::make('Project Phaseable', 'projectPhaseable')->types([
                Estate::class,
                Family::class,
                Property::class,
            ])->default(Estate::class),
        ];
    }
```

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Setup a polymorphic relationship as per the example here in the docs: https://laravel.com/docs/10.x/eloquent-relationships#many-to-many-polymorphic-relations
2. Add BelongsTo field in you Taggable resource

Thanks in advance!

Luke

**Comment by crynobone** [comment]
`BelongsTo::make('Phase', 'projectPhaseable', Phase::class),`

Bad relationship name here.

**Comment by lukejwatts** [comment]
> `BelongsTo::make('Phase', 'projectPhaseable', Phase::class),`
> 
> Bad relationship name here.

So should it be `BelongsTo::make('Phase', 'projectPhase', Phase::class)` or something else?

**Comment by crynobone** [comment]
![SmartSelect_20240723_093238_Chrome](https://github.com/user-attachments/assets/8673a5c3-2c5e-4b4b-b9b5-a1ce3b40088f)

From the documentation , and you can also refer to https://laravel.com/docs/11.x/eloquent-relationships#one-to-one-defining-the-inverse-of-the-relationship
